opal/mca/common/ucx/common_ucx.c

```diff
@@ -416,8 +416,10 @@ OPAL_DECLSPEC int opal_common_ucx_mca_pmix_fence(ucp_worker_h worker)
         return ret;
     }
 
-    while (!fenced) {
-        ucp_worker_progress(worker);
+    MCA_COMMON_UCX_PROGRESS_LOOP(worker) {
+        if(fenced) {
+            break;
+        }
```

Comment:
seems the difference if whether progress is called if fenced==true before line 419?
but if fence is already completed why still need to call progress?

Reply:
no, the fix here is that `MCA_COMMON_UCX_PROGRESS_LOOP` also invokes `opal_progress` which progresses hcoll